tokio/src/runtime/time/entry.rs

```diff
@@ -543,6 +543,17 @@ impl TimerEntry {
         }
     }
 
+    pub(crate) fn reset_without_reregister(mut self: Pin<&mut Self>, new_time: Instant) {
```

Comment:
This function is very similar to `reset`. Maybe `reset` should take an argument `reregister: bool`. I'm not 100% sure, but it is worth considering given the `unsafe` in the fn.

Reply:
Ah thanks, good point. I checked and there were not too many usages of `TimerEntry::reset` to update, so I added the boolean flag and removed `reset_without_reregister`.